app/models/host/managed.rb

```diff
@@ -434,15 +434,13 @@ def info
     info_hash
   end
 
-  # JSON is auto-parsed by the API, so these should be in the right format
-  def self.import_host_and_facts(hostname, facts, certname = nil, proxy_id = nil)
+  def self.import_host(hostname, import_type, certname = nil, proxy_id = nil)
```

Comment:
Can you add default option to `import_type = "puppet"` instead of explicitly assigning it?